lib/web_console/session.rb

```diff
@@ -59,11 +59,44 @@ def eval(input)
     #
     # Returns nothing.
     def switch_binding_to(index)
-      @evaluator = Evaluator.new(@bindings[index.to_i])
+      @evaluator = Evaluator.new(@current_binding = @bindings[index.to_i])
+    end
+
+    # Returns context of the current binding
+    def context(obj)
+      ( object_name?(obj) ? context_of(obj) : global_context ).flatten
     end
 
     private
 
+      def object_name?(s)
+        s.is_a?(String) && !s.empty? && !s.match(/[^a-zA-Z0-9\@\$\.\:]/)
```

Comment:
> Do we need to be that defensive here (the type check)?

It seems no need. I will drop the regular expression and keep others as inline conditions for type checking